ui/page/dexclient/add_dex_widget.go

```diff
@@ -0,0 +1,181 @@
+package dexclient
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"decred.org/dcrdex/client/asset"
+	"decred.org/dcrdex/client/core"
+	"gioui.org/layout"
+	"gioui.org/widget"
+	"github.com/planetdecred/dcrlibwallet"
+	"github.com/planetdecred/godcr/ui/decredmaterial"
+	"github.com/planetdecred/godcr/ui/load"
+	"github.com/planetdecred/godcr/ui/modal"
+	"github.com/planetdecred/godcr/ui/page/components"
+	"github.com/planetdecred/godcr/ui/values"
+)
+
+const testDexHost = "dex-test.ssgen.io:7232"
+
+type addDexWidget struct {
+	*load.Load
+	addDexServer     decredmaterial.Button
+	dexServerAddress decredmaterial.Editor
+	isSending        bool
+	cert             decredmaterial.Editor
+	createWalletWdg  *dexCreateWalletWidget
+}
+
+func newAddDexWidget(l *load.Load) *addDexWidget {
+	dwg := &addDexWidget{
+		Load:             l,
+		dexServerAddress: l.Theme.Editor(new(widget.Editor), "DEX Address"),
+		addDexServer:     l.Theme.Button("Submit"),
+		cert:             l.Theme.Editor(new(widget.Editor), "Cert content"),
+	}
+
+	dwg.addDexServer.TextSize = values.TextSize12
+	dwg.addDexServer.Background = l.Theme.Color.Primary
+
+	dwg.dexServerAddress.Editor.SingleLine = true
+	if l.WL.MultiWallet.NetType() == dcrlibwallet.Testnet3 {
+		dwg.dexServerAddress.Editor.SetText(testDexHost)
+	}
+
```

Comment:
Not sure why we are using this here.
<img width="798" alt="image" src="https://user-images.githubusercontent.com/27733432/144828342-2a75de73-e629-4332-a763-f046282026fa.png">
I dont see a cancel button. What if i change my mind?
Do we really need this? I believe the text_input_modal.go can be modified to use 2 input fields and used.

or you can create a better widget modal for this in the dex package just like it was done for staking purchace modal.
https://github.com/planetdecred/godcr/blob/master/ui/page/staking/purchase_modal.go